app/models/foreman_tasks/task/search.rb

```diff
@@ -22,6 +22,15 @@ def search_by_taxonomy(key, operator, value)
         sql = "foreman_tasks_locks_taxonomy#{uniq_suffix}.resource_id #{operator} ? OR foreman_tasks_locks_taxonomy#{uniq_suffix}.resource_id IS NULL"
         { :conditions => sanitize_sql_for_conditions([sql, value]), :joins => joins }
       end
+
+      # Expects the time in the format "\d+ (seconds|minutes|hours|days|months|years)"
+      SUPPORTED_DURATION_FORMAT = /\A\s*(\d+\s+\b(seconds?|minutes?|hours?|days?|months?|years?)\b)\b\s*\z/i
+      def search_by_duration(key, operator, value)
+        raise "Unsupported duration '#{value}' specified for searching" unless value =~ SUPPORTED_DURATION_FORMAT
+        value = value.strip
+        { :conditions => "coalesce(ended_at, current_timestamp) - coalesce(coalesce(started_at, ended_at), current_timestamp) #{operator} ?::interval",
```

Comment:
the `::interval` may very well be a postgresism and we may need to find a workaround for sqlite

Reply:
tests are failing on sqlite because of this